pkg/cluster/context.go

```diff
@@ -101,7 +102,11 @@ func (c *Context) ClusterName() string {
 
 // Create provisions and starts a kubernetes-in-docker cluster
 func (c *Context) Create(cfg *config.Config, retain bool, wait time.Duration) error {
-	// validate config first
+	// default config fields (important for usage as a library, where the config
+	// may be constructed in memory rather than from disk)
+	encoding.Scheme.Default(cfg)
```

Comment:
I don't totally love what all is going on here, but to use kind as a library we need to apply defaulting etc on supplied config because we don't know that it went through load etc.